src/main/java/yapper/Parser.java

```diff
@@ -0,0 +1,317 @@
+        }
+    }
+
+    /**
+     * Parses task list data to create and add tasks to the {@link TaskList}.
+     *
+     * @param data Task list data saved in the local file represented as a String.
+     * @throws YapperException Data found in local file is detected to be incorrectly formatted.
+     */
+    public void parseData(String data) throws YapperException {
+        String[] taskData = data.split("" + " / ");
+        boolean isDone; // used for creating new Task
+        switch (taskData[0]) {
+        case "T":
+            if (taskData.length != 3) {
+                throw new YapperException("Error in the save files 1");
+            }
+
+            if (taskData[1].equals("0")) {
+                isDone = false;
+            } else if (taskData[1].equals("1")) {
+                isDone = true;
+            } else {
+                throw new YapperException("Error in the save files 2");
+            }
+
+            mainTasks.addTaskNoMessage(new Todo(isDone, taskData[2]));
+            break;
+        case "D":
+            if (taskData.length != 4) {
+                throw new YapperException("Error in the save files 3");
+            }
+
+            if (taskData[1].equals("0")) {
+                isDone = false;
+            } else if (taskData[1].equals("1")) {
+                isDone = true;
+            } else {
+                throw new YapperException("Error in the save files 4");
+            }
+
+            try {
+                LocalDate by = LocalDate.parse(taskData[3]);
+                mainTasks.addTaskNoMessage(new Deadline(isDone, taskData[2], by));
+            } catch (DateTimeParseException e) {
+                throw new YapperException("Error in the save files 5");
+            }
+
+            break;
+        case "E":
+            if (taskData.length != 5) {
+                throw new YapperException("Error in the save files 6");
+            }
+
+            if (taskData[1].equals("0")) {
+                isDone = false;
+            } else if (taskData[1].equals("1")) {
+                isDone = true;
+            } else {
+                throw new YapperException("Error in the save files 7");
+            }
+            try {
+                LocalDate from = LocalDate.parse(taskData[3]);
+                LocalDate to = LocalDate.parse(taskData[3]);
+                mainTasks.addTaskNoMessage(new Event(isDone, taskData[2], from, to));
+            } catch (DateTimeParseException e) {
+                throw new YapperException("Error in the save files 8");
+            }
+            break;
+        default:
+            throw new YapperException("Error in the save files 9");
+        }
+    }
+
+    /**
+     * Parses all tasks in the {@link TaskList} into format to be saved in the local file.
+     *
+     * @return String representation of task list data to be saved to the local file.
+     */
+    public String parseToData() {
```

Comment:
slight nit; perhaps the names for `parseData()` and `parseToData()` can be made clearer e.g `parseInputData()`, `parseOutputData()`?